src/inc/til/some.h

```diff
@@ -0,0 +1,214 @@
+
+        void fill(const _Ty& _Value)
+        {
+            array::fill(_Value);
+        }
+
+        void swap(some& _Other) noexcept(_Is_nothrow_swappable<_Ty>::value)
+        {
+            array::swap(_Other);
+            std::swap(_Used, _Other._Used);
+        }
+
+        _NODISCARD _CONSTEXPR17 iterator begin() noexcept
+        {
+            return iterator(_Elems, 0);
+        }
+
+        _NODISCARD _CONSTEXPR17 const_iterator begin() const noexcept
+        {
+            return const_iterator(_Elems, 0);
+        }
+
+        _NODISCARD _CONSTEXPR17 iterator end() noexcept
+        {
+            return iterator(_Elems, _Used);
+        }
+
+        _NODISCARD _CONSTEXPR17 const_iterator end() const noexcept
+        {
+            return const_iterator(_Elems, _Used);
+        }
+
+        _NODISCARD _CONSTEXPR17 reverse_iterator rbegin() noexcept
+        {
+            return reverse_iterator(end());
+        }
+
+        _NODISCARD _CONSTEXPR17 const_reverse_iterator rbegin() const noexcept
+        {
+            return const_reverse_iterator(end());
+        }
+
+        _NODISCARD _CONSTEXPR17 reverse_iterator rend() noexcept
+        {
+            return reverse_iterator(begin());
+        }
+
+        _NODISCARD _CONSTEXPR17 const_reverse_iterator rend() const noexcept
+        {
+            return const_reverse_iterator(begin());
+        }
+
+        _NODISCARD _CONSTEXPR17 const_iterator cbegin() const noexcept
+        {
+            return begin();
+        }
+
+        _NODISCARD _CONSTEXPR17 const_iterator cend() const noexcept
+        {
+            return end();
+        }
+
+        _NODISCARD _CONSTEXPR17 const_reverse_iterator crbegin() const noexcept
+        {
+            return rbegin();
+        }
+
+        _NODISCARD _CONSTEXPR17 const_reverse_iterator crend() const noexcept
+        {
+            return rend();
+        }
+
+        _NODISCARD constexpr size_type size() const noexcept
+        {
+            return _Used;
+        }
+
+        _NODISCARD constexpr size_type max_size() const noexcept
+        {
+            return _Size;
+        }
+
+        _NODISCARD constexpr bool empty() const noexcept
+        {
+            return !_Used;
+        }
+
+        _NODISCARD _CONSTEXPR17 reference at(size_type _Pos)
+        {
+            if (_Used <= _Pos)
+            {
+                _Xran();
+            }
+
+            return _Elems[_Pos];
+        }
+
+        _NODISCARD constexpr const_reference at(size_type _Pos) const
+        {
+            if (_Used <= _Pos)
+            {
+                _Xran();
+            }
+
+            return _Elems[_Pos];
+        }
+
+        _NODISCARD _CONSTEXPR17 reference operator[](_In_range_(0, _Used - 1) size_type _Pos) noexcept /* strengthened */
+        {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Pos < _Used, "some subscript out of range");
+#endif // _CONTAINER_DEBUG_LEVEL > 0

```

Comment:
if you compose an `array`, accessing _its_ `operator[]` will properly throw so we don't take a dependency on these private details

Reply:
OK.